File: example/go/config/Equip/Jewelryrandom.go
package equip

type Jewelryrandom struct {
    lvlRank config.LevelRank //等级
    attackRange config.Range //最小攻击力
    otherRange []config.Range //最小防御力
    testPack []config.Equip.TestPackBean //测试pack
}

//getters
func (t *Jewelryrandom) GetLvlRank() config.LevelRank {
    return t.lvlRank
}

func (t *Jewelryrandom) GetAttackRange() config.Range {
    return t.attackRange
}

func (t *Jewelryrandom) GetOtherRange() []config.Range {
    return t.otherRange
}

func (t *Jewelryrandom) GetTestPack() []config.Equip.TestPackBean {
    return t.testPack
}

var all []Jewelryrandom
func GetAll() []Jewelryrandom {
    return all[:len(all)]:len(all)]
}

var allMap map[config.LevelRank]Jewelryrandom
func Get(key config.LevelRank) (Jewelryrandom,bool){
    return allMap[key]
}
        public static Jewelryrandom Get(config.LevelRank lvlRank)
        {
            Jewelryrandom v;
            return all.TryGetValue(lvlRank, out v) ? v : null;
        }

        public static List<Jewelryrandom> All()
        {
            return all.OrderedValues;
        }

        public static List<Jewelryrandom> Filter(Predicate<Jewelryrandom> predicate)
        {
            var r = new List<Jewelryrandom>();
            foreach (var e in all.OrderedValues)
            {
                if (predicate(e))
                    r.Add(e);
            }
            return r;
        }

        internal static void Initialize(Config.Stream os, Config.LoadErrors errors)
        {
            all = new Config.KeyedList<config.LevelRank, Jewelryrandom>();
            for (var c = os.ReadInt32(); c > 0; c--) {
                var self = _create(os);
                all.Add(self.LvlRank, self);
            }
        }

        internal static void Resolve(Config.LoadErrors errors) {
            foreach (var v in All())
                v._resolve(errors);
        }

File: example/java/config/equip/Jewelryrandom.java
package config.equip;

public class Jewelryrandom {
    private config.LevelRank lvlRank;
    private config.Range attackRange;
    private java.util.List<config.Range> otherRange;
    private java.util.List<config.equip.TestPackBean> testPack;

    private Jewelryrandom() {
    }

    public static Jewelryrandom _create(configgen.genjava.ConfigInput input) {
        Jewelryrandom self = new Jewelryrandom();
        self.lvlRank = config.LevelRank._create(input);
        self.attackRange = config.Range._create(input);
        {
            int c = input.readInt();
            if (c == 0) {
                self.otherRange = java.util.Collections.emptyList();
            } else {
                self.otherRange = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.otherRange.add(config.Range._create(input));
                }
            }
        }
        {
            int c = input.readInt();
            if (c == 0) {
                self.testPack = java.util.Collections.emptyList();
            } else {
                self.testPack = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.testPack.add(config.equip.TestPackBean._create(input));
                }
            }
        }
        return self;
    }

    /**
     * 等级
     */
    public config.LevelRank getLvlRank() {
        return lvlRank;
    }

    /**
     * 最小攻击力
     */
    public config.Range getAttackRange() {
        return attackRange;
    }

    /**
     * 最小防御力
     */
    public java.util.List<config.Range> getOtherRange() {
        return otherRange;
    }

    /**
     * 测试pack
     */
    public java.util.List<config.equip.TestPackBean> getTestPack() {
        return testPack;
    }

    @Override
    public String toString() {
        return "(" + lvlRank + "," + attackRange + "," + otherRange + "," + testPack + ")";
    }

    public void _resolve(config.ConfigMgr mgr) {
        lvlRank._resolve(mgr);
    }

    public static Jewelryrandom get(config.LevelRank lvlRank) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.getEquipJewelryrandom(lvlRank);
    }

    public static java.util.Collection<Jewelryrandom> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.allEquipJewelryrandom();
    }

    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            for (int c = input.readInt(); c > 0; c--) {
                Jewelryrandom self = Jewelryrandom._create(input);
                mgr.equip_jewelryrandom_All.put(self.lvlRank, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            for (Jewelryrandom e : mgr.equip_jewelryrandom_All.values()) {
                e._resolve(mgr);
            }
        }

    }

}
